etc/contracts-test-data/contracts/custom-account/validation-rule-breaker.sol

```diff
@@ -0,0 +1,138 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+pragma solidity ^0.8.0;
+
+import "./Constants.sol";
+import "./TransactionHelper.sol";
+
+import "./SystemContractsCaller.sol";
+
+import "./interfaces/IAccount.sol";
+
+contract ValidationRuleBreaker is IAccount {
+    using TransactionHelper for Transaction;
+
+    uint32 public typeOfRuleBreak;
+
+    constructor() {
+        typeOfRuleBreak = 0;
+    }
+
+    function setTypeOfRuleBreak(uint32 _typeOfRuleBreak) external {
+        typeOfRuleBreak = _typeOfRuleBreak;
+    }
+
+    function validateTransaction(
+        bytes32 _txHash,
+        bytes32 _suggestedSignedTxHash,
+        Transaction calldata _transaction
+    ) external payable override returns (bytes4 magic) {
+        // By default we consider the transaction as successful
+        magic = VALIDATION_SUCCESS_MAGIC;
+
+        if (typeOfRuleBreak == 1) {
+            // The balance of another account may not be read
+            // I'm writing assertions because otherwise the compiler would optimize away the access
+            require(BOOTLOADER_FORMAL_ADDRESS.balance != 0);
+        } else if (typeOfRuleBreak == 2) {
+            // May not call an EOA
+            address(1234567890).call("");
+        } else if (typeOfRuleBreak == 3) {
+            // TODO make test that calls upgradeable proxy
+        } else if (typeOfRuleBreak == 100) {
```

Comment:
- Do you plan to add this in this PR? If not, makes sense to remove this for now and restore when the test is added.
- What about reading allowed / disallowed storage slots?

Reply:
Yes, I want to add this today. The upgradeable proxy is one case of allowed storage slots.